utils/utils.go

```diff
@@ -0,0 +1,126 @@
+package utils
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"io"
+	"os"
+	"strconv"
+	"net/url"
+	"reflect"
+	"bootcamp/domain/model"
+	"github.com/spf13/viper"
+	"gopkg.in/mgo.v2/bson"
+)
+
+/*
+GetEnvVar gets a given envar by key
+*/
+func GetEnvVar(key string) (string, error) {
+  viper.SetConfigFile(".env")
+  err := viper.ReadInConfig()
+  value, ok := viper.Get(key).(string)
+
+  if !ok {
+    err = errors.New("Invalid type assertion")
+  }
+
+  return value, err
+}
+
+/*
+GetObjectIdFromParams transforms an /{id} to a ObjectId
+*/
+func GetObjectIdFromParams(params map[string]string) (bson.ObjectId, error) {
+	var objectId bson.ObjectId
+	id := params["id"]
+
+	if id == "" || !bson.IsObjectIdHex(id) {
+		return objectId,	errors.New("Invalid id provided")
+	}
+
+	objectId = bson.ObjectIdHex(id)
+	return objectId, nil
+}
+
+/*
+GetPokemonFromReader decode Pokemon from reader (body request) and returns the Pokemon information
+*/
+func GetPokemonFromReader(reader io.ReadCloser) (model.Pokemon, error) {
+	var tempPokemon model.Pokemon
+	decoder := json.NewDecoder(reader)
+	err := decoder.Decode(&tempPokemon)
+
+	if err == nil {
+		defer reader.Close()		
+	}
+
+	return tempPokemon, err
+}
+
+/*
+ReadCSV read a CSV with Pokemon information and transform the content to the Pokemon struct type
+*/
+func ReadCSV() (model.PokemonList, error) {
+	var pokemonList model.PokemonList
+	recordFile, err := os.Open("assets/pokemon.csv")
+
+	if err == nil {
```

Comment:
You can avoid indenting so much if you do the following

```
if err != nil {
  return nil, err
}

for ... { ... }
```

Reply:
*